src/worker/controller/actions/sync/AnnounceLocalStateAction.js

```diff
@@ -10,52 +10,43 @@ class AnnounceLocalStateAction {
     this._controller = controller;
   }
   execute(params) {
-    const useCache = params.cache;
     const onResponse = params.onResponse;
     let isEngCid = params.isEngCid;
 
-    if (useCache) {
-      this._controller.cache().getAllTips((err, tipsList) => {
-        // TODO:: implement cache logic
-        // TODO:: if cache empty still query core since maybe it was deleted or first time
-      });
-    } else {
-      this._controller.execCmd(constants.NODE_NOTIFICATIONS.GET_ALL_ADDRS, {
-        useCache: useCache,
-        onResponse: (err, allAddrsResponse) => {
-          /**
-           * do the announcement
-           * */
-          const parsedEngCids = allAddrsResponse.result.addresses
-            .map(addr => {
-              const ecid = EngCid.createFromSCAddress(addr);
-              if (ecid) {
-                return ecid;
-              } else {
-                this._controller.logger().error("[-] err converting bytearray->hex->EngCid !");
-              }
-            })
-            .filter(ecid => {
-              return ecid !== undefined && ecid !== null;
-            });
-          isEngCid = true;
-          this._controller.provider().provideContentsBatch(parsedEngCids, isEngCid, (err, failedCids) => {
-            if (err) {
-              // TODO:: this is completley incorrect.
-              // TODO:: it shows like there was some total error, but there will be errrors in the case where 1 peer logged out
-              // TODO:: it will try to reconnect to him in the DHT and will throw an error
-              // TODO:: this is ok accestable behaviour.
-              // TODO:: the log below is missleading it says that there was a general error but that's no true, everything still works/
-              // TODO:: Bottom line im not processing the errors in the correct way.
-              return onResponse(err, parsedEngCids);
+    this._controller.execCmd(constants.NODE_NOTIFICATIONS.GET_ALL_ADDRS, {
+      onResponse: (err, allAddrsResponse) => {
+        /**
+         * do the announcement
+         * */
+        const parsedEngCids = allAddrsResponse.result.addresses
+          .map(addr => {
+            const ecid = EngCid.createFromSCAddress(addr);
+            if (ecid) {
+              return ecid;
             } else {
-              this._controller.logger().debug("[+] success providing cids.");
-              return onResponse(null, parsedEngCids);
+              this._controller.logger().error("[-] err converting bytearray->hex->EngCid !");
             }
+          })
+          .filter(ecid => {
+            return ecid !== undefined && ecid !== null;
           });
-        }
-      });
```

Comment:
This is a pretty weird usage of `map` and `filter`.
Are we sure this is the behavior we want? In case of an error just ignore it and log it?

Reply:
Removed the `map` and the `filter` but the error is still only shown in the logs